### DISCUSSION TITLE
Casting types

### DISCUSSION THREAD

**Discussion by NoahELE** [microsoft/pyright#4799] [Q&A] [answered]
```python
import pandas as pd


def foo(i: int, v: int) -> None:
    return


s = pd.Series([1, 2, 3, 4, 5])
for i, v in s.items():
    i: int # error here
    foo(i, v)

```

I try to tell the type checker that i should be an int, but it doesn't work.

What should I do to cast i to int?

> I have `"python.analysis.typeCheckingMode": "basic"` setting in vscode.

**Comment by soraros** [comment, marked as answer]
You could use `typing.cast` to inform the type checker. Suppose `i: int` and `v: float`, one might write

```python
from typing import cast

for i, v in s.items():
    i = cast(int, i)
    v = cast(float, v)
    foo(i, v)
```

or

```python
for i, v in cast(Iterator[tuple[int, float]], s.items()):
    foo(i, v)
```

If runtime check is desirable, `asset isinstance(x, A)` also refines `x`'s type to `A`

```python
for i, v in s.items():
    assert isinstance(i, int)
    foo(i, v)
```

**Reply by NoahELE** [reply]
Thanks!
